File: 2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int[] res = new int[nums.length];
        int even_i = 0;
        int odd_i = 1;
        for (int num : nums) {
            if (num > 0) {
                res[even_i] = num;
                even_i += 2;
            } else {
                res[odd_i] = num;
                odd_i += 2;
            } 
        }
        return res;
    }
}

File: 2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        res = [0] * len(nums)
        e_i, o_i = 0, 1
        for num in nums:
            if num > 0:
                res[e_i] = num
                e_i += 2
            else:
                res[o_i] = num
                o_i += 2
        return res